src/contracts/mocks/MockStakeHelper.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {Aludel} from "alchemist/contracts/aludel/Aludel.sol";
+import {AludelV2} from "../aludel/AludelV2.sol";
+
+contract MockStakeHelper {
+    function flashStake(
+        address geyser,
+        address vault,
+        uint256 amount,
+        bytes calldata lockPermission,
+        bytes calldata unstakePermission
+    ) external {
+        Aludel(geyser).stake(vault, amount, lockPermission);
```

Comment:
I don't know about this particular usecase, but shouldn't there be a hook to do something in between the stake and unstake?

Reply:
You mean to simulate a flash-loan behavior?